### PR TITLE
Added an event to viewtopic.php to modify template vars.

### PR DESCRIPTION
**Pull request by DerDave**
[feature/event]
(This is my first pull requst, I hope I don't screw this up)
I added an event to viewtopic.php tomodify the template vars as described in the ticket.

PHPBB3-13480

### CODE REVIEW

File: phpBB/viewtopic.php [hunk 1]
```diff
@@ -712,9 +712,32 @@
...
+/**
+* Event to modify template variables before assigning
+*
+* @event core.viewtopic_assign_template_vars_before
+* @var	array	page_data			Viewtopic page-data that will be parsed 
+* 									via $template->assign_vars()
+* @var	int		forum_id			Forum ID
+* @var	array	topic_data			Array with topic data
+* @var	int		topic_id			Topic ID
+* @var	int		total_posts			Topic total posts count
```
Comment (marc1706) [inline phpBB/viewtopic.php]: $total_posts doesn't exist in the vars array below.

File: phpBB/viewtopic.php [hunk 2]
```diff
@@ -712,9 +712,32 @@
...
+* @since 3.2
```
Comment (marc1706) [inline phpBB/viewtopic.php]: Any reason you chose 3.2? Otherwise it should be 3.1.3-RC1 right now.

File: phpBB/viewtopic.php [hunk 3]
```diff
@@ -712,9 +712,32 @@
...
+	'page_data',
+	'forum_id',
+	'topic_data',
+	'topic_id',
+);
+extract($phpbb_dispatcher->trigger_event('core.viewtopic_assign_template_vars', compact($vars)));
+
+$template->assign_vars($page_data);
+
+
```
Comment (marc1706) [inline phpBB/viewtopic.php]: There shouldn't be two consecutive new lines. Remove one of them.

**Comment by marc1706** [comment]
Please modify your commit message to fit into this format: https://wiki.phpbb.com/Git#Commit_Messages

File: phpBB/viewtopic.php (hunk 2 above)
Comment (nickvergessen) [inline phpBB/viewtopic.php]: Should be `3.1.4-RC1`

**Comment by DerDave** [comment]
How do I do this? Should I simpl make this pull request again?

regards, Dave

**Comment by nickvergessen** [comment]
If you use the console, try `git commit --amend`

**Comment by DerDave** [comment]
No, I use Tortoise... So ist it okay, to just start another pull request.

What exactly is wrong with the commit message? I tried to use the mentioned format?

What are "consecutive new lines"? Two blank lines?

**Comment by rxu** [comment]
With TortoiseGit it's even simplier - just choose ```Git Commit -> "develop"```, then check the ```"Amend last commit"``` checkbox, edit the commit message and then perform commit, then make ```"Git Sync..."``` having the ```"Force"``` checkbox checked.

**Comment by nickvergessen** [comment]
> What exactly is wrong with the commit message? I tried to use the mentioned format?

it should be
```
[ticket/13480] Added an event to viewtopic.php to modify template vars.

PHPBB3-13480
```
> What are "consecutive new lines"? Two blank lines?

yes

**Comment by DerDave** [comment]
So... I tried again...

File: phpBB/viewtopic.php [hunk 4]
```diff
@@ -712,9 +712,30 @@
...
+* @var	int		topic_id			Topic ID
+* @since 3.1.4-RC1
+*/
+$vars = array(
+	'page_data',
+	'forum_id',
+	'topic_data',
+	'topic_id',
+);
+extract($phpbb_dispatcher->trigger_event('core.viewtopic_assign_template_vars', compact($vars)));
```
Comment (marc1706) [inline phpBB/viewtopic.php]: Should have the same event name as in the docblock above.

**Comment by nickvergessen** [comment]
@DerDave want to try fixing this, or shall we take it over?

**Comment by DerDave** [comment]
Sorry, I've just accepted a new job... I don't have much time right now....

**Comment by marc1706** [comment]
Replaced by https://github.com/phpbb/phpbb/pull/3368

**Pull request closed without merging**